docs/pages/UPGRADE-4.0.md

```diff
@@ -0,0 +1,7 @@
+# Upgrade 4.0
+
+## Subscription
+
+* The constructor of the `SubscriptionEngine` class has been changed
+* * Instead of passing a `Store` instance, you now need to pass a `MessageLoader` instance.
+* * Instead of passing a `RetryStrategy` instance, you now need to pass a `RetryStrategyRepository` instance.
```

Comment:
I think we should be a little more presice here about how to upgrade besides using the other interface(s).

Like mentioning that there is a `StoreMessageLoader` which takes a `Store` as an argument e.g.